python_modules/jupedsim/jupedsim/distributions.py

```diff
@@ -1,2 +1,440 @@
+        with neighbouring coordinates illustrated below: I.e. those cells that could
+        contain points closer than r.
+
+                                         ooo
+                                        ooooo
+                                        ooXoo
+                                        ooooo
+                                         ooo
+
+        """
+        dxdy = [(-1, -2), (0, -2), (1, -2), (-2, -1), (-1, -1), (0, -1), (1, -1), (2, -1),
+                (-2, 0), (-1, 0), (1, 0), (2, 0), (-2, 1), (-1, 1), (0, 1), (1, 1), (2, 1),
+                (-1, 2), (0, 2), (1, 2), (0, 0)]
+        neighbours = []
+        for dx, dy in dxdy:
+            neighbour_coords = coords[0] + dx, coords[1] + dy
+            if not (0 <= neighbour_coords[0] < self.nx and
+                    0 <= neighbour_coords[1] < self.ny):
+                # We're off the grid: no neighbours here.
+                continue
+            neighbour_cell = self.cells[neighbour_coords]
+            if neighbour_cell is not None:
+                # This cell is occupied: store this index of the contained point.
+                neighbours.append(neighbour_cell)
+        return neighbours
+
+    def get_neighbours(self, pt):
+        coords = self.get_cell_coords(pt)
+        return self.determine_neighbours(coords)
+
+
+class GridMock(Grid):
+    def __init__(self):
+        pass
+
+
+def intersecting_area_polygon_circle(mid_point, radius, polygon):
+    """returns the intersecting area of circle and polygon"""
+    # creates a point
+    point = shply.Point(mid_point)
+    # creates a layer with the size of the radius all around this point
+    circle = point.buffer(radius)
+    # creates a polygon in shapely
+    poly = shply.Polygon(polygon)
+    # returns the size of the intersecting area
+    return poly.intersection(circle).area
+
+
+def is_inside_circle(point, mid, min_r, max_r):
+    """checks if a point is inside a Circle segment reaching from minimum radius to maximum radius"""
+    dif_x = point[0] - mid[0]
+    dif_y = point[1] - mid[1]
+    circle_equation = dif_x ** 2 + dif_y ** 2
+    return min_r ** 2 <= circle_equation <= max_r ** 2
+
+
+def get_bounding_box(polygon):
+    """returns an Axis Aligned Bounding Box containing the minimal/maximal x and y values
+    formatted like : [(min(x_values), min(y_values)), (max(x_values), max(y_values))]"""
+    x_values, y_values = [], []
+    for point in polygon:
+        x_values.append(point[0])
+        y_values.append(point[1])
+
+    return [(min(x_values), min(y_values)), ((max(x_values)), max(y_values))]
+
+
+def min_distance_to_polygon(pt, polygon):
+    """returns the minimal distance between a point and every line segment of a polygon"""
+    shapely_polygon = polygon[:]
+    shapely_polygon.append(polygon[0])
+    return shply.LineString(shapely_polygon).distance(shply.Point(pt))
+
+
+def create_random_points(polygon, count, agent_radius, wall_distance,
```

Comment:
This should be two functions:
First: Fill polygon with density
Second: Fill polygon with num agents

Reply:
done